psl-core/src/main/java/org/linqs/psl/model/rule/Rule.java

```diff
@@ -38,4 +38,14 @@
 	public int groundAll(AtomManager atomManager, GroundRuleStore groundRuleStore);
 
 	public boolean isWeighted();
+
+	/**
+	 * @return a name for the rule
+	 */
+	public String getName();
+
+	/**
+	 * @param name a name for the rule
+	 */
+	public void setName(String name);
```

Comment:
Is there a reason to allow rules to be renamed?
Can we just have it set at construction?